File: plugins/exists.py
import re
from plugin import Plugin


class ExistsPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'存在する\s+(.+)', line)

        if not match:
            return None

        name = match.group(1)

        return f"print({name!r} in globals())"

File: web/plugins/exists.js
import { WebPlugin } from "../engine/plugin.js";

export class ExistsPlugin extends WebPlugin {
    constructor() {
        super("存在する");
    }

    async execute(line, context) {
        const expression = line
            .replace(/^存在する\s*/, "");

        const name = expression
            .replace(/^「|」$/g, "")
            .trim();

        context.runtime.write(
            context.runtime.exists(name)
        );
    }
}